### DISCUSSION TITLE
Filter one column by value in second column

### DISCUSSION THREAD

**Discussion by rahuldubey11** [supabase/supabase#5053] [Questions]
Hello,

Is it possible to run a query with filters on another column's value, like:

```
final res = await supabase
  .from('cities')
  .select('*')
  .eq('columnA', 'columnB')
  .execute(); 
```

**Comment by silentworks** [comment]
I don't think you can do this, even in normal SQL you would have to run a subquery to select that column from the same table to compare it. Can you state what you are trying to accomplish with this, please?

**Comment by rahuldubey11** [comment]
Thank you for the response, let me describe it in details.

I have 2 tables:

**1. users**
<table>
  <tr>
    <th>id</th>
    <th>lastStoryTime</th> 
  </tr>
  <tr>
    <td>user1</td>
    <td>123</td> 
  </tr>
  <tr>
    <td>user2</td>
    <td>456</td> 
  </tr>
</table>

**2. storiesViewed**
<table>
  <tr>
    <th>id</th>
    <th>viewerID</th>
    <th>ownerID</th>
    <th>viewedTimeLastStory</th> 
  </tr>
  <tr>
    <td>1</td>
    <td>user1</td>
    <td>user2</td>
    <td>567</td> 
  </tr>
  <tr>
    <td>2</td>
    <td>user2</td>
    <td>user1</td>
    <td>789</td> 
  </tr></table>

What I need is, all the rows from the 'users' table where:
```
final response = await supabase
        .from('users')
        .select(*assume this is set up correctly using foreign keys*)
        .gt('lastStoryTime', 'storiesViewed.viewedTimeLastStory')
        .execute();
```
